### ISSUE TITLE
[BUG] WebWorker is still typescript in dist folder. IE11 Support.

### ISSUE THREAD

**Issue by nawlbergs** [angular/angular#44836]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

No

### Description

# tldr:
- im need to support IE11 (so still on v12)
- the web-worker file generated via cli is still a `.ts` file inside my dist folder (IE no likey)
- the docs that that show you how to get your web-worker url using `include.meta` did not return the correct path in IE11.

# What happened
1. Using https://v12.angular.io/guide/web-worker
2. I added web-worker via cli to my poject.. it added a tsconfig file.. updated my angular.json etc.
3. I moved my webworker code into a service and updated paths..
    ` Worker = new Worker(new URL('./../super-logic.worker', import.meta.url));`
4. Everything ran fine in chrome.. but now IE11 was throwing `Object doesn't not support this action`.

# What  was wrong

### First Issue
After farting around for a few hours (adding shims.. looking at compiled code etc etc).. i realized the actual file in the dist folder was not `.js` but still a `.ts` file!  The url being provided to the Worker via (`new URL('./../super-logic.worker', import.meta.url)`) was `...../survey/super-logic.ts`.. and obviously IE11 didnt like that. (`/survey` is my dist/route)

**My Fix**: Changed the worker to a `js` file and edited the `tsconfig.webworker.ts` to only include `.js` files:
```
  "include": [
    "src/**/*.worker.js"
  ]
```

### Second Issue
During my debugging... I added console logs to my service to print out the url to the service worker.  For some reason, IE11 was producing a different URL than what Chrome was.
- Chrome: `...mydomain.local/survey/8230219b602f9582c8b7.js`
- IE11: `...mydomain.local/survey/take/8230219b602f9582c8b7.js`
So again.. obviously IE could not find the file...   The service was provided in root.. but it might be important to note that this service was being injected from a module which was being lazy-loaded.

**My Fix**:  Since all the files are dumped in root of the dist folder... I just hard-coded the path to the dist folder into service and used the generated filename provided by the new URL:
```javascript
      const url = new URL('./../../shared/workers/logic-worker.worker', import.meta.url);
      // IE11 hack.. for some reason this is wrong.. just force to dist folder root
      const finalUrl = `/survey/${url.toString().split('/').pop()}`;
      this.worker = new Worker(finalUrl);
```

Hopefully this helps others with this issue if nothing else.  Thanks.

### Please provide a link to a minimal reproduction of the bug

https://github.com/nawlbergs/ng-webworker-bug

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 12.1.4
Node: 12.22.9
Package Manager: npm 7.24.2
OS: win32 x64

Angular: 12.1.5
... animations, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1201.4
@angular-devkit/build-angular   12.1.4
@angular-devkit/core            12.1.4
@angular-devkit/schematics      12.1.4
@angular/cli                    12.1.4
@schematics/angular             12.1.4
ng-packagr                      12.2.6
rxjs                            6.6.7
typescript                      4.3.5
```

**Comment by nawlbergs** [comment]
So i just added a webworker in a fresh 12.1.4 project and the webworker compiles to `js`.  I dont know wtf is wrong with our project... ill see if i can dig into this more later.

**Comment by AndrewKushnir** [comment]
Closing this ticket for now. Please reopen if needed.

**Comment by nawlbergs** [comment]
Now im truly confused.

1. i check out my branch again
2. ng build... is still not compling worker to js.. its the source `ts` file sitting in my dist folder (tho a file comparison to original shows some line spacing changes)
3. i wanted to get it back to a working state to figure out the issue... started comparing everything to the starter app i made to test earlier...  commenting out code... added a second worker..  re-building... removing ie support etc... tons of `ng build`... until it finally compiled to js.
4. I shelved my changes to save the state of a working webworker so i could start to roll things back slowly and see what changes caused this to occur. 
5. BUT... now my project is back to what it was when i checked it out.. however.... the web worker is, for some reason... now (still) compiling to js....   hummm?  so now its working.. even tho now i have zero code changes.

now im just not trusting this.  perhaps there is some strange caching going on with npm, ts or angular that is borked.

I deleted node_modules..  `npm verify cache` `npm install` ... now I cannot get it to break.  Im a little scared to roll this to production (what if our pipeline suddenly starts to not compile it)

**Comment by nawlbergs** [comment]
@AndrewKushnir Can you re-open this please?

https://github.com/nawlbergs/ng-webworker-bug

so...  after many hours...
```
   // this works and compiles to ts
   const worker = new Worker(new URL('./app.worker', import.meta.url));  // this works and compiles to ts.

   // does not work.. this file remains ts in dist folder
   const url = new URL('./app.worker', import.meta.url);
   const worker = new Worker(url);
```
